openmc/model/triso.py

```diff
@@ -153,3 +162,843 @@ def create_triso_lattice(trisos, lower_left, pitch, shape, background):
+
+        Parameters
+        ----------
+        i : int
+            Index in particles array of particle for which to find nearest
+            neighbor.
+
+        Returns
+        -------
+        int
+            Index in particles array of nearest neighbor of i
+        double
+            distance between i and nearest neighbor.
+
+        """
+
+        # Need the second nearest neighbor of i since the nearest neighbor
+        # will be itself. Using argpartition, the k-th nearest neighbor is
+        # placed at index k.
+        idx = list(mesh[cell_index(particles[i])])
+        dists = cdist([particles[i]], particles[idx])[0]
+        if dists.size > 1:
+            j = dists.argpartition(1)[1]
+            return idx[j], dists[j]
+        else:
+            return None, None
+
+
+    def update_rod_list(i, j):
+        """Update the rod list with the new nearest neighbors of particles i
+        and j since their overlap was eliminated.
+
+        Parameters
+        ----------
+        i, j : int
+            Index of particles in particles array.
+
+        """
+
+        # If the nearest neighbor k of particle i has no nearer neighbors,
+        # remove the rod currently containing k from the rod list and add rod
+        # k-i, keeping the rod list sorted
+        k, d_ik = nearest(i)
+        if k and nearest(k)[0] == i:
+            remove_rod(k)
+            add_rod(d_ik, i, k)
+        l, d_jl = nearest(j)
+        if l and nearest(l)[0] == j:
+            remove_rod(l)
+            add_rod(d_jl, j, l)
+
+        # Set inner diameter to the shortest distance between two particle
+        # centers
+        if rods:
+            inner_diameter[0] = rods[0][0]
+
+
+    def cell_index_cube(p, cl=None):
+        """Calculate the index of the lattice cell in which the given particle
+        center falls.
+
+        Parameters
+        ----------
+        p : list of float
+            Cartesian coordinates of particle center.
+        cl : list of float
+            Length of the lattice cells in x-, y-, and z-directions.
+
+        Returns
+        -------
+        tuple of int
+            Indices of lattice cell.
+
+        """
+
+        if cl is None:
+            cl = cell_length
+
+        return tuple(int(p[i]/cl[i]) for i in range(3))
+
+
+    def cell_index_cylinder(p, cl=None):
+        """Calculate the index of the lattice cell in which the given particle
+        center falls.
+
+        Parameters
+        ----------
+        p : list of float
+            Cartesian coordinates of particle center.
+        cl : list of float
+            Length of the lattice cells in x-, y-, and z-directions.
+
+        Returns
+        -------
+        tuple of int
+            Indices of lattice cell.
+
+        """
+
+        if cl is None:
+            cl = cell_length
+
+        return tuple([int((p[0] + domain_radius)/cl[0]),
+                      int((p[1] + domain_radius)/cl[1]), int(p[2]/cl[2])])
```

Comment:
Looks like you could just construct the tuple directly. No need to create a list and then cast.